Demo/RCHSearchViewController.swift

```diff
@@ -13,22 +13,57 @@ private let reuseIdentifier = "productCell"
 class RCHSearchViewController: UIViewController {
 
     @IBOutlet weak var searchBar: UISearchBar!
+    @IBOutlet weak var searchBarView: UIView!
     @IBOutlet weak var searchProductsView: UIView!
     @IBOutlet weak var searchResultsCollectionView: UICollectionView!
     @IBOutlet weak var searchProductsLabel: UILabel!
     @IBOutlet weak var searchProductsImageView: UIImageView!
     @IBOutlet weak var autocompleteTableView: UITableView!
+    @IBOutlet weak var sortFilterView: UIView!
+    @IBOutlet weak var sortButton: UIButton!
+    @IBOutlet weak var filterButton: UIButton!
+    
+    var nextButton = UIButton()
+    var previousButton = UIButton()
+    var sortPicker = UIPickerView()
+    var pickerView = UIView()
     
     weak var timer = Timer()
+    var currentSort: SortingOptions = .relevance {
+        didSet {
+            pickerView.isHidden = true
+            let titleString = "Sort: \(currentSort.rawValue)"
+            sortButton.setTitle(titleString, for: .normal)
+            searchForProducts()
+        }
+    }
     
+    var pageCount = 0 {
+        didSet {
+            if pageCount > 0 {
+                previousButton.isEnabled = true
+                previousButton.setTitleColor(.blue, for: .normal)
+            } else {
+                previousButton.isEnabled = false
+                previousButton.setTitleColor(.gray, for: .normal)
+            }
+        }
+    }
+    
+    var pageStart = 0 {
+        didSet {
+           searchForProducts()
+        }
+    }
```

Comment:
I'm not sure I get the difference between pageCount and pageStart. pageStart is duplicated state that could be computed with `pageCount / 20`.